pkg/exec/action.go

```diff
@@ -7,23 +7,35 @@ import (
 var _ builder.ExecutableAction = Action{}
 
 type Action struct {
+	Name  string
 	Steps []Step // using UnmarshalYAML so that we don't need a custom type per action
 }
 
+func (a Action) MarshalYAML() (interface{}, error) {
+	return map[string]interface{}{a.Name: a.Steps}, nil
+}
+
+func (a Action) MakeSteps() interface{} {
+	return &[]Step{}
+}
+
 // UnmarshalYAML takes any yaml in this form
 // ACTION:
 // - exec: ...
 // and puts the steps into the Action.Steps field
 func (a *Action) UnmarshalYAML(unmarshal func(interface{}) error) error {
-	var steps []Step
-	results, err := builder.UnmarshalAction(unmarshal, &steps)
+	results, err := builder.UnmarshalAction(unmarshal, a)
```

Comment:
This is a breaking change to that function signature. Mixins won't be affected unless they were using it, for example the kuberentes mixin doesn't but the az mixin does. They don't need to update and use this change unless they want to do linting or mixin configuration. This helps them unmarshal the action data for that.

Reply:
I have created a follow-up issue in the skeletor repo to update and use this change: https://github.com/deislabs/porter-skeletor/issues/16